numpy/_core/tests/test_strings.py

```diff
@@ -490,3 +490,138 @@ def test_strip_args(self, in1, in2, out, dt):
         in2 = np.array(in2, dtype=dt)
         out = np.array(out, dtype=dt)
         assert_array_equal(np.strings.strip(in1, in2), out)
+
+    @pytest.mark.parametrize("in1,in2,in3,in4,out", [
```

Comment:
Not having meaningful names makes this test less readable.

Also, it seems there are quite a lot of cases that effectively are duplicates (basically, from the "here and there" one onward).

Reply:
They're not exact duplicates, but they might be testing the same path. I'd prefer to be on the safe side, and test this as intensively as possible, even if it is a bit more than necessary, but I'm okay to also remove some cases if you feel that's better.

Reply:
I don't mind having extra test cases, as long as the tests run quickly.

Reply:
Sure, no worries. Though as noted, none of these test arrays or broadcasting. Those are covered by `test_defchararray`, but perhaps good to have them in one place?

Reply:
I've moved the tests that test arrays and broadcasting over to `test_strings` as well. IMO it's okay as is, the base `test_replace` really just tests the `replace` logic and the others test the "wrapper".